exec/cnex/global.c

```diff
@@ -1218,6 +1218,46 @@ void object__invokeMethod(TExecutor *exec)
         goto cleanup;
     }
 
+    if (obj->object->type == oArray) {
+        if (strcmp(name, "size") == 0) {
+            if (args->array != NULL && args->array->size != 0) {
+                exec->rtl_raise(exec, "DynamicConversionException", "invalid number of arguments to size() (expected 0)");
+                goto cleanup;
+            }
+            push(exec->stack, cell_fromObject(object_createNumberObject(number_from_uint64(((Cell*)obj->object->ptr)->array->size))));
+            goto cleanup;
+        }
+    }
+
+    if (obj->object->type == oDictionary) {
+        if (strcmp(name, "size") == 0) {
+            if (args->array != NULL && args->array->size != 0) {
+                exec->rtl_raise(exec, "DynamicConversionException", "invalid number of arguments to size() (expected 0)");
+                goto cleanup;
+            }
+            push(exec->stack, cell_fromObject(object_createNumberObject(number_from_sint64(((Cell*)obj->object->ptr)->dictionary->len))));
+            goto cleanup;
+        } else if (strcmp(name, "keys") == 0) {
+            if (args->array != NULL && args->array->size != 0) {
+                exec->rtl_raise(exec, "DynamicConversionException", "invalid number of arguments to keys() (expected 0)");
+                goto cleanup;
+            }
+            // ToDo: Consider doing an "inplace" conversion, where the "keys" Cell is manually converted into an Object Array.
```

Comment:
What does this comment mean? What would be the benefit of changing the way this works now?

Reply:
That was a mental note to myself, to see if I could maybe keep the same Cell Array, but convert all the members to Object Strings, from TStrings, without having to build a whole new array.  I wanted to write the code to try it out, but then realized I was optimizing a problem I didn't even know I had; so, I left it as a comment, and walked away.